rear_obstacle_checker_node:
  common:
    on_time_buffer:
      type: double
      validation:
        gt_eq<>: [0.0]
    off_time_buffer:
      type: double
      validation:
        gt_eq<>: [0.0]
    pointcloud:
      object_buffer:
        type: double
        validation:
          gt_eq<>: [0.0]
    range:
      pointcloud:
        forward:
          type: double
          validation:
            gt_eq<>: [0.0]
        backward:
          type: double
          validation:
            gt_eq<>: [0.0]
      object:
        forward:
          type: double
          validation:
            gt_eq<>: [0.0]
        backward:
          type: double
          validation:
            gt_eq<>: [0.0]
    prediction:
      ego:
        reaction_time:
          type: double
          validation:
            gt<>: [0.0]
        max_acceleration:
          type: double
          validation:
            gt<>: [0.0]
        max_deceleration:
          type: double
          # validation:
          #   lt<>: [0.0]
        max_positive_jerk:
          type: double
          validation:
            gt<>: [0.0]
        max_negative_jerk:
          type: double
          # validation:
          #   lt<>: [0.0]
      object:
        reaction_time:
          type: double
          validation:
            gt<>: [0.0]
        max_deceleration:
          type: double
          # validation:
          #   lt<>: [0.0]

    predicted_path:
      min_velocity:
        type: double
      max_velocity:
        type: double
      acceleration:
        type: double
      time_resolution:
        type: double
      time_horizon:
        type: double
      delay_until_departure:
        type: double
    rss:
      extended_polygon_policy:
        type: string
      vehicle_deceleration:
        type: double
      vehicle_reaction_time:
        type: double
      vehicle_safety_time_margin:
        type: double
      lateral_distance_max_threshold:
        type: double
      longitudinal_distance_min_threshold:
        type: double
      longitudinal_velocity_delta_time:
        type: double

  scene:
    type: string_array
    default_value:
      [
        static_obstacle_avoidance,
        lane_change_left,
        lane_change_right,
        start_planner,
        goal_planner,
        behavior_path_planner,
      ]
    validation:
      unique<>: []
      subset_of<>:
        [
          [
            static_obstacle_avoidance,
            lane_change_left,
            lane_change_right,
            start_planner,
            goal_planner,
            blind_spot,
            behavior_path_planner,
          ],
        ]

  __map_scene:
    topic:
      type: string
    behavior:
      type: string_array
      validation:
        unique<>: []
        not_empty<>: []
        subset_of<>: [[slow_down, stop, shift_left, shift_right, turn_left, turn_right]]
    safety_condition:
      type: string
      validation:
        one_of<>: [[safe, unsafe, both]]
    check_control_points:
      type: bool
    target_type:
      type: string_array
      validation:
        unique<>: []
        not_empty<>: []
        subset_of<>:
          [[bicycle, bus, car, motorcycle, pedestrian, trailer, truck, unknown, pointcloud]]
    current_lane:
      type: bool
    adjacent_lane:
      type: bool
